microsoft/vscode
# Suggest uses aria-describedby for description

This change hides the description content from screen readers, and instead uses this information as the aria-labelledby element for the current list item. This change addresses #167900. The way this is accomplished is:

* The list item description element receives a label id based on the current list item id.
* The description element id is provided to the 'aria-labelledby' attribute of the list item.
* The description element is hidden so it is not announced as part of the content of the currently selected list item.

This changes the screen reader output from:

> DisplayPositionclass DisplayPosition, Selected (1 of 1)

To:

> DisplayPosition, Selected (1 of 1), class DisplayPosition

To test this change:

1. open a local vscode browser instance using `scripts/code-web.sh`
2. enable a screen reader (JAWS, NVDA, VoiceOver, Talkback)
3. in the browser editor window, in a typescript file, use the keys cmd + space to open the suggest listbox
4. use the keyboard to navigate the list. Observe that the spoken suggestion contains only the suggestion as the label, and announces the description after the identifying information has been announced.

[aria-labelledby-suggest.webm](https://user-images.githubusercontent.com/13710580/229942696-f19f97e3-9f9e-4a10-afe2-468cd9ec8b7e.webm)
